lib/iris/tests/unit/util/test_array_equal.py

```diff
@@ -44,9 +44,14 @@ def test_nd(self):
         self.assertTrue(array_equal(array_a, array_b))
         self.assertFalse(array_equal(array_a, array_c))
 
-    def test_masked_is_ignored(self):
+    def test_masked_is_not_ignored(self):
         array_a = ma.masked_array([1, 2, 3], mask=[1, 0, 1])
         array_b = ma.masked_array([2, 2, 2], mask=[1, 0, 1])
+        self.assertTrue(array_equal(array_a, array_b))
+
+    def test_masked_is_different(self):
```

Comment:
This is good stuff, but I'd like us also to be clear what happens when you compare masked arrays with "regular" ones.
E.G. :
   * array([1,0]) == ma.masked_array([1, 0])
   * array([1,0]) == ma.masked_array([1, 0], mask=[0, 0])  # (not necessarily the same as mask=False)
   * array([1,0]) != ma.masked_array([1, 0], mask=[0, 1])